tests/unit/module_utils/core/test_scrub_none_parameters.py

```diff
@@ -1,56 +1,94 @@
+     True,
      dict(param1='something', param4='something_else')
      ),
     (dict(param1=dict(sub_param1='something', sub_param2=None), param2=None, param3=None, param4='something_else'),
+     True,
      dict(param1=dict(sub_param1='something'), param4='something_else')
      ),
     (dict(param1=dict(sub_param1='something', sub_param2=dict(sub_sub_param1='another_thing')), param2=None, param3=None, param4='something_else'),
+     True,
      dict(param1=dict(sub_param1='something', sub_param2=dict(sub_sub_param1='another_thing')), param4='something_else')
      ),
     (dict(param1=dict(sub_param1='something', sub_param2=dict()), param2=None, param3=None, param4='something_else'),
+     True,
      dict(param1=dict(sub_param1='something', sub_param2=dict()), param4='something_else')
      ),
     (dict(param1=dict(sub_param1='something', sub_param2=False), param2=None, param3=None, param4='something_else'),
+     True,
      dict(param1=dict(sub_param1='something', sub_param2=False), param4='something_else')
      ),
     (dict(param1=None, param2=None),
+     True,
      dict()
      ),
     (dict(param1=None, param2=[]),
+     True,
      dict(param2=[])
+     ),
+    (dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2='my_dict_nested_in_a_list_2')], param2=[]),
+     True,
+     dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2='my_dict_nested_in_a_list_2')], param2=[])
+     ),
+    (dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2='my_dict_nested_in_a_list_2')], param2=[]),
+     False,
+     dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2='my_dict_nested_in_a_list_2')], param2=[])
+     ),
+    (dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2=None)], param2=[]),
+     True,
+     dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1')], param2=[])
+     ),
+    (dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2=None)], param2=[]),
+     False,
+     dict(param1=[dict(sub_param1='my_dict_nested_in_a_list_1', sub_param2=None)], param2=[])
+     ),
+    (dict(param1=[dict(sub_param1=[dict(sub_sub_param1=None)], sub_param2=None)], param2=[]),
+     True,
+     dict(param1=[dict(sub_param1=[{}])], param2=[])
+     ),
+    (dict(param1=[dict(sub_param1=[dict(sub_sub_param1=None)], sub_param2=None)], param2=None),
+     False,
+     dict(param1=[dict(sub_param1=[dict(sub_sub_param1=None)], sub_param2=None)]),
      )
 ]
 
 
-@pytest.mark.parametrize("input_params, output_params", scrub_none_test_data)
-def test_scrub_none_parameters(input_params, output_params):
+@pytest.mark.parametrize("input_params, descend_into_lists_flag, output_params", scrub_none_test_data)
+def test_scrub_none_parameters(input_params, descend_into_lists_flag, output_params):
 
-    assert scrub_none_parameters(input_params) == output_params
+    assert scrub_none_parameters(input_params, descend_into_lists_flag) == output_params
```

Comment:
I think we'd get better coverage if this was 

```
@pytest.mark.parametrize("input_params, output_params_descend, output_params_nodescend", scrub_none_test_data) 
def test_scrub_none_parameters(input_params, output_params_descend, output_params_nodescend):
    assert scrub_none_parameters(input_params) == output_params_descend
    assert scrub_none_parameters(input_params, True) == output_params_descend
    assert scrub_none_parameters(input_params, False) == output_params_nodescend
```
a) We test the default behaviour - will cause the unit tests to flag things up if someone changed the defaults
b) We test that the descend code didn't have some weird side effect we didn't expect